webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/servers/PolicyServerManagementService.scala

```diff
@@ -416,3 +438,136 @@ object RelaySynchronizationMethod {
           }
   }
 }
+
+
+/*
+ * System directives / rules / etc for a policy server.
+ */
+final case class PolicyServerConfigurationObjects(
+    directives: Map[TechniqueName, Directive]
+  , groups    : List[NodeGroup]
+  , targets   : List[PolicyServerTarget]
+  , rules     : List[Rule]
+)
+
+object PolicyServerConfigurationObjects {
+  val v1_0: TechniqueVersion = TechniqueVersion.parse("1.0").getOrElse(throw new RuntimeException(s"Error in default version data, this is likely a bug. Please report it."))
+
+  implicit class ToDirective(id: String) {
+    def toDirective = Directive(
+        DirectiveId(DirectiveUid(id))
+      , v1_0
+      , Map ()
+      , id
+      , s""
+      , None
+      , ""
+      , 0
+      , _isEnabled = true
+      , isSystem = true
+      , Tags(Set.empty)
+    )
+  }
+
+  val relayTechniques = List("server-common", "rudder-service-apache","rudder-service-relayd")
+  val rootTechniques = List("rudder-service-postgresql","rudder-service-slapd","rudder-service-webapp")
+
+  def directiveCommonHasPolicyServer(nodeId: NodeId, hostname: String, policyServerId: NodeId) = {
```

Comment:
this signature is super confusing: i would expect the policyServerId to be the final policyServerId (so the relay id) - i think it needs either a comment or better method naming